lib/chef/provider/package.rb

```diff
@@ -84,17 +100,18 @@ def action_install
       end
 
       def action_upgrade
-        if candidate_version.nil?
+        if !candidate_version_array.any?
           Chef::Log.debug("#{@new_resource} no candidate version - nothing to do")
+          return
         elsif @current_resource.version == candidate_version
           Chef::Log.debug("#{@new_resource} is at the latest version - nothing to do")
-        else
-          @new_resource.version(candidate_version)
-          orig_version = @current_resource.version || "uninstalled"
-          converge_by("upgrade package #{@new_resource.package_name} from #{orig_version} to #{candidate_version}") do
-            upgrade_package(@new_resource.package_name, candidate_version)
-            Chef::Log.info("#{@new_resource} upgraded from #{orig_version} to #{candidate_version}")
-          end
+          return
+        end
+        @new_resource.version(candidate_version)
```

Comment:
yeah... but I'm trying to keep this as close to the original as possible and then you can make further fixes. I want to get this merged